sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -808,7 +809,14 @@ class Column(val expr: Expression) extends Logging {
    * @group expr_ops
    * @since 2.4.0
    */
-  def isInCollection(values: scala.collection.Iterable[_]): Column = isin(values.toSeq: _*)
+  def isInCollection(values: scala.collection.Iterable[_]): Column = withExpr {
+    val hSet = values.toSet[Any]
```

Comment:
we always create the set to deduplicate the values?

Reply:
We can move it to the if condition.